### ISSUE TITLE
dotted lines appear solid in exported pdf

### ISSUE THREAD

**Issue by hiobsch** [djoshea/matlab-save-figure#3]
Hi there,
I find the issue that dotted lines in the Matlab-plot appear as solid lines in the exported pdf-file.
The issue has been discussed a lot in the savefigure-project too. From there I took the idea to change the Matlab-Renderer (set(gcf,'Renderer','painters')) but that did not proof successful (neither did the zbuffer-Renderer.
For any hints I would be very thankful since the software looks to work very well appart from this issue (quick processing, very good looking latex-text representation)

**Comment by djoshea** [comment]
Hi there, thanks for reporting this. Glad you've found this useful! It's still astonishing one has to jump through all these hoops to get a decent figure out of Matlab :)

There are two different "workflows" inside saveFigure, depending on the complexity of the figure, which parallels Matlab's internal way of saving figures (if you dig through print.m you can find the references to matlab.graphics.internal.PrintVertexChecker and checker.exceedsLimits()). The gist is that Matlab uses painters when the figure is relatively "simple", and if it's complex it falls back on a rasterized format. In saveFigure, I check the same complexity threshold, and either save to an SVG (simple case -dsvg) or an EPS (complex case -despc2). You'll notice the difference because complex figures will be rasterized even when saved to vector formats. Anyway, the complex case saves as EPS and might be the source of your issue.

Unfortunately, I'm not able to reproduce it. If I do:

``` matlab
clf;
plot(randn(200, 2000), ':');
saveFigure('test');
```

![test](https://cloud.githubusercontent.com/assets/77752/16662843/5a1d4322-442e-11e6-8482-fa3e53a7f8ef.png)

I see dotted lines.

It could be dependent on the platform and versions though. Do you want to try downloading export_fig and using Oliver Woodford's `print2eps` to do the eps export instead? This might solve your problem, and produces identical results for me. This would be as simple as changing out line 271 in saveFigure.m

``` matlab
%    print(hfig, sprintf('-r%d', resolution), '-depsc', file);
    print2eps(file, hfig, struct('fontswap', false), sprintf('-r%d', resolution));
```

If this fixes things for you, I'll just swap this in since it makes no difference on my end. Let me know! Thanks.

**Comment by hiobsch** [comment]
Hi djoshea,

Thank you for your detailed answer.
Yes it is astonishing that it is still complicated to consistently 
produce good-looking figures with Matlab taking into consideration its 
price. But on the other hand graphics simply is something rather 
complicated.

I have noticed that Matlab falls back onto rasterized graphic export if 
the figure contains a very large amount of points -- which unfortunately 
often is the case in simulations. I still have to figure out how I will 
deal with all those issues in the end, I consider using pgfplots and 
reducing the points in my graphs beforehand.

Anyway to come back to the problem I reported on: I should have been a 
little more clear. The problem occurs in 3d-plots when the line is 
running in a direction close to perpendicular to the screen or 
projection plane. I only tested one case for now but your proposal 
didn't change anything just there, see the pdfs in the attachments.

Anyway I want to thank you for developing and maintaining SaveFigure! 
Keep up the good work :-)

Jakob
